content/articles/sentiment-analysis-with-naive-bayes-algorithms/index.md

```diff
@@ -0,0 +1,1035 @@
+author: qoyum-olatunde-yusuf
+date: 2021-10-13T00:00:00-15:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+- url: /engineering-education/sentiment-analysis-with-naive-bayes-algorithms/hero.PNG
+
+alt: sentiment analysis with naive bayes algorithms cover image
+
+---
+Many scholars have defined sentiment analysis; hardly there exists a new definition for the term that has not been heard of before. On that note, I will say sentiment analysis is the systematic method of extracting emotions with machine learning. This tutorial will show how to implement sentiment analysis using naive Bayes algorithms.
+
+<!--more-->
+
+Sentiment analysis has been helpful in many fields, in discovering hate speech, analyzing public opinion about a product or brand, building chatbots, etc.
+
+
+### Table of contents
+
+1. [Prerequisites](#prerequisites)
+2. [Goal](#goal)
+3. [Why Naive Bayes Models?.](#why-naive-bayes-models)
+4. [IMDB movie review dataset](#imdb-movie-review-dataset)
+5. [Installing necessary libraries](#installing-necessary-libraries)
+6. [Function to process the IMDB datasets](#function-to-process-the-imdb-datasets)
+7. [Preparing the notebook for BernoulliNB](#preparing-the-notebook-for-bernoullinb)
+8. [Preparing the notebook for CategoricalNB](#preparing-the-notebook-for-categoricalnb)
+9. [Preparing the notebook for ComplementNB](#preparing-the-notebook-for-complementnb)
+10. [Preparing the notebook for GaussianNB](#preparing-the-notebook-for-gaussiannb)
+11. [Preparing the notebook for MultinomialNB](#preparing-the-notebook-for-multinomialnb)
+12. [Conclusion](#conclusion)
+13. [Additional resources](#additional-resources)
+
+### Prerequisites
+
+First and foremost, we want to build a sentiment analysis project using `Bayesian models` and discover which of its algorithms is best. Also, it is not a work of which algorithm is best for sentiment analysis, but rather, which of the `naive Bayes algorithms` is best for this project. I firmly believe my readers are machine learning students who are expected to be versed in basic Python programming and must be good at performing data analysis with [Pandas](https://pandas.pydata.org/) and [Numpy](https://numpy.org/).
+
+### Goal
+
+At the end of this tutorial, readers should be able to:
+
+- understand how sentiment analysis works,
+
+- build a supervised machine learning model that predicts the sentiments of a word or sentence given to it,
+
+- determine which is better between count vectorizer and tfidf vectorizer,
+
+- compare the naive Bayes model that is best for text classification, and
+
+- use Pandas functions for data analysis.
+
+
+
+
+### Why Naive Bayes Models?.
+
+The Bayesian theorem is the core principle behind Naive Bayes algorithms. It assumes that all features are independent of one another. Thus, we can as well say that the classifier believes that one particular attribute in a class does not influence the presence of another. 
```

Comment:
The independence notion is incorrect. The data samples are independent of each other, not the attributes of the data. Please correct the entire paragraph.